### PR TITLE
Replace CategoricalMLPRegressor

### PR DESCRIPTION
**Pull request by zequnyu**
- Deleted CategoricalMLPRegressor
- Renamed CategoricalMLPRegressorWithModel to CategoricalMLPRegressor and did minor style/name update
- Updated test accordingly

[Transition test](https://github.com/rlworkgroup/garage/blob/catmlp_regressor_transition_test/tests/garage/tf/regressors/test_categorical_mlp_regressor_with_model_transit.py)

[Result](https://travis-ci.com/rlworkgroup/garage/builds/128789446)

No integration test and benchmarking needed.

### CODE REVIEW

File: src/garage/tf/regressors/__init__.py [hunk 1]
```diff
@@ -21,8 +17,7 @@
 from garage.tf.regressors.gaussian_mlp_regressor import GaussianMLPRegressor
 
 __all__ = [
-    'BernoulliMLPRegressor', 'BernoulliMLPRegressorWithModel',
```
Comment (ahtsan) [inline src/garage/tf/regressors/__init__.py]: Why `BernoulliMLPRegressorWithModel` is also removed?

Comment (zequnyu) [reply src/garage/tf/regressors/__init__.py]: Might be a mistake...

**Comment by ryanjulian** [comment]
Can you explain why benchmarks are not needed?

**Comment by zequnyu** [comment]
> Can you explain why benchmarks are not needed?

As discussed with Anson, this should be similar to [BernoulliMLPRegressor](https://github.com/rlworkgroup/garage/pull/890#issue-319992039) right?

**Comment by ryanjulian** [comment]
Ah I see. Yes, it doesn't appear benchmarks are needed.

**Pull request merged**